nemo/collections/nlp/metrics/qa_metrics.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+import json
+import re
+import string
+import torch
+
+from nemo.collections.nlp.parts.utils_funcs import tensor2list
+
+
+class QAMetrics(object):
+    @staticmethod
+    def remove_articles(text):
+        return re.sub(r"\b(a|an|the)\b", " ", text)
+
+    @staticmethod
+    def white_space_fix(text):
+        return " ".join(text.split())
+
+    @staticmethod
+    def remove_punc(text):
+        exclude = set(string.punctuation)
+        return "".join(ch for ch in text if ch not in exclude)
+
+    @staticmethod
+    def normalize_answer(s: str):
+        """ Lower text and remove punctuation, articles and extra whitespace """
+
+        return QAMetrics.white_space_fix(QAMetrics.remove_articles(QAMetrics.remove_punc(s.lower())))
+
+    @staticmethod
+    def _get_normalized_tokens(s: str):
+        """ Get normalized tokens """
+        if not s:
```

Comment:
nit: I think your code might handle the base case already? `''.split()` returns []

Reply:
SInce the str passes through the normalize answer method before performing split, checking `not s` condition avoids error in normalize answer method